### PR TITLE
WIP: enable log dumping for Windows test nodes

### PR DESCRIPTION
**Pull request by pjh**
Updates log-dump.sh for Windows nodes.

Tested:
```
$ PROJECT=${CLOUDSDK_CORE_PROJECT} KUBERNETES_SKIP_CONFIRM=y NUM_NODES=2 \
  NUM_WINDOWS_NODES=2 KUBE_GCE_ENABLE_IP_ALIASES=true go run \
  ./hack/e2e.go -- --up
$ cluster/log-dump/log-dump.sh
$ ls _artifacts
```

### CODE REVIEW

**Comment by pjh** [comment]
@yujuhong I think this is pretty much working. The first couple commits do some prerequisite repairs/cleanups and the final commit updates log-dump.sh. PTAL when you have a chance; I'll incorporate feedback, clean it up and then send a PR upstream so we can get this activated for our new upstream test job.

cc @mtaufen

File: cluster/gce/util.sh [hunk 1]
```diff
@@ -2965,7 +2961,9 @@ function kube-down() {
     # change during a cluster upgrade.)
     local templates=$(get-template "${PROJECT}")
 
-    for group in ${INSTANCE_GROUPS[@]:-}; do
+    local all_instance_groups=(${INSTANCE_GROUPS[@]} ${WINDOWS_INSTANCE_GROUPS[@]})
+    echo "PJH: all_instance_groups[@]: ${all_instance_groups[@]}"  # REMOVE
```
Comment (yujuhong) [inline cluster/gce/util.sh]: Per your comment, remove this? :D

File: hack/ginkgo-e2e.sh [hunk 2]
```diff
@@ -100,6 +100,8 @@ fi
 # set before we get here, which eliminates any cluster/gke use if
 # KUBERNETES_CONFORMANCE_PROVIDER is set to "gke".
 if [[ -z "${NODE_INSTANCE_GROUP:-}" ]] && [[ "${KUBERNETES_PROVIDER}" == "gke" ]]; then
+  # TODO(pjh): this code is broken, detect-node-instance-groups appears nowhere
```
Comment (yujuhong) [inline hack/ginkgo-e2e.sh]: ........

File: cluster/skeleton/util.sh [hunk 3]
```diff
@@ -28,11 +28,6 @@ function detect-node-names {
 	echo "NODE_NAMES: [${NODE_NAMES[*]}]" 1>&2
 }
 
-# Get node IP addresses and store in KUBE_NODE_IP_ADDRESSES[]
```
Comment (yujuhong) [inline cluster/skeleton/util.sh]: Why deleting this from skeleton?

File: cluster/log-dump/log-dump.sh [hunk 4]
```diff
@@ -321,6 +381,7 @@ function find_non_logexported_nodes() {
   succeeded_nodes=$(gsutil ls ${gcs_artifacts_dir}/logexported-nodes-registry) || return 1
   echo "Successfully listed marker files for successful nodes"
   NON_LOGEXPORTED_NODES=()
+  # TODO(pjh): check WINDOWS_NODE_NAMES here?
```
Comment (yujuhong) [inline cluster/log-dump/log-dump.sh]: Create a k8s issue:`Support dumping logs with log exporter for GCE windows nodes`
and reference that issue here.

File: cluster/centos/util.sh [hunk 5]
```diff
@@ -47,15 +47,6 @@ function detect-masters() {
   echo "KUBE_MASTER_IP_ADDRESSES: [${KUBE_MASTER_IP_ADDRESSES[*]}]" 1>&2
 }
 
-# Get node IP addresses and store in KUBE_NODE_IP_ADDRESSES[]
```
Comment (yujuhong) [inline cluster/centos/util.sh]: Why deleting this function?

Comment (pjh) [reply cluster/centos/util.sh]: The first commit in this PR removes some code that I'm pretty sure is unused. I came across it while making my other changes. Would it be worth fixing this upstream as part of my PR, or as a separate one?

File: cluster/skeleton/util.sh (hunk 3 above)
Comment (pjh) [reply cluster/skeleton/util.sh]: See other comment - there are no callers of detect-nodes() in the entire tree.

File: hack/ginkgo-e2e.sh (hunk 2 above)
Comment (pjh) [reply hack/ginkgo-e2e.sh]: Should I file an issue about this and then I or someone can fix it later?

I'm pretty sure this code doesn't work... cc @mtaufen

Comment (yujuhong) [reply hack/ginkgo-e2e.sh]: You can file an issue or send a separate PR later to remove the code.

File: cluster/skeleton/util.sh (hunk 3 above)
Comment (yujuhong) [reply cluster/skeleton/util.sh]: I'd prefer sending a separate cleanup PR later.

File: cluster/log-dump/log-dump.sh (hunk 4 above)
Comment (pjh) [reply cluster/log-dump/log-dump.sh]: I filed https://github.com/kubernetes/kubernetes/issues/74439, please add comments there if it doesn't capture what you meant for this.

**Comment by pjh** [comment]
Thanks for the suggestions, I've incorporated them and sent https://github.com/kubernetes/kubernetes/pull/74438 upstream.

**Pull request closed without merging**